tools/ci_build/build.py

```diff
@@ -448,14 +448,25 @@ def parse_arguments():
                         "To enable support for custom operators pass 'custom_ops' as a parameter. "
                         "e.g. '--minimal_build custom_ops'. This can be combined with an 'extended' build by passing "
                         "'--minimal_build extended custom_ops'")
-    parser.add_argument("--include_ops_by_config", type=str,
-                        help="include ops from config file. "
-                        "See /docs/Reduced_Operator_Kernel_build.md for more information.")
-    parser.add_argument("--enable_reduced_operator_type_support", action='store_true',
-                        help='If --include_ops_by_config is specified, and the configuration file was created from ORT '
-                             'format models with type reduction enabled, limit the types individual operators support '
-                             'where possible to further reduce the build size. '
-                             'See /docs/Reduced_Operator_Kernel_build.md for more information.')
+
+    parser.add_argument(
+        "--include_ops_by_config", type=str,
+        help="Include ops from config file. "
+        "See /docs/Reduced_Operator_Kernel_build.md for more information.")
+
+    reduced_ops_exclusive_args = parser.add_mutually_exclusive_group()
+    reduced_ops_exclusive_args.add_argument(
+        "--enable_reduced_operator_type_support", action='store_true',
+        help='If --include_ops_by_config is specified, and the configuration file was created from ORT '
+        'format models with type reduction enabled, limit the types individual operators support '
+        'where possible to further reduce the build size. '
+        'Mutually exclusive with --reduced_operator_allowed_types. '
+        'See /docs/Reduced_Operator_Kernel_build.md for more information.')
+    reduced_ops_exclusive_args.add_argument(
+        "--reduced_operator_allowed_types", nargs="*", metavar="type",
+        help="If --include_ops_by_config is specified, constrain the types individual operator kernels support to "
+        "these allowed types. "
+        "Mutually exclusive with --enable_reduced_operator_type_support.")
 
     parser.add_argument("--disable_contrib_ops", action='store_true',
```

Comment:
What's the expected usage here? It seems somewhat onerous to specify a long list of types in the build command line, especially so if you have a config file with required ops (this combination is the primary expected usage).  

Would it be simpler usage to do everything via a single config file for the types and ops? When processing the config if you have a global types entry we ignore any op specific types. Either option requires `--enable_reduced_operator_type_support` to be specified.

If the config file only contains a global types entry we would use that to reduce type support in all ops. We need to maintain current behavior that if the config file does not specify any required ops we disable all ops, as we rely on that to test the base build size.

Reply:
we can put globally allowed types in the configuration file too. if it's there, then i think also having op specific types in the same file is confusing and maybe we shouldn't consider that valid.

Reply:
I'm fine with rejecting the config file as invalid if it has both per-op types and a global type list.